### DISCUSSION TITLE
Does SWR cache persist on page reload ?

### DISCUSSION THREAD

**Discussion by jeremytenjo** [vercel/swr#592] [General] [answered]
Does SWR cache persist on page reload ?

**Comment by sergiodxa** [comment, marked as answer]
It doesn't, it's all in-memory, but you can use [swr-sync-storage](https://github.com/sergiodxa/swr-sync-storage) to keep your SWR cache synced with localStorage or sessionStorage and it will hydrate your cache from there when your app load.

**Reply by jeremytenjo** [reply]
You should have linked your article explaining how to use it https://dev.to/sergiodxa/sync-swr-cache-with-web-storage-8fp. It's great

**Reply by demian85** [reply]
Is there a way to configure how much "space" the SWR cache uses? Will it grow forever? It's weird there is no documentation related to the cache, it's like a black box.

**Reply by sergiodxa** [reply]
There is no limit right now, but there is an open PR to add garbage collection of unused cache keys so it could be limited.
